pkg/repository/v1_repository.go

```diff
@@ -427,7 +427,14 @@ func (r *V1Repository) updateComponentManifest(id string) (*v1manifest.Component
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	item, ok := index.Components[id]
+	var components map[string]v1manifest.ComponentItem
+	if withYanked {
+		components = index.ComponentListWithYanked()
+	} else {
+		components = index.ComponentList()
+	}
+
+	item, ok := components[id]
 	if !ok {
 		return nil, errUnknownComponent
```

Comment:
The returned error should contain call stack.
```suggestion
		return nil, errors.AddStack(errUnknownComponent)
```